### ISSUE TITLE
[Bug] tpp-run -print option does not print kernel output correctly

### ISSUE THREAD

**Issue by nhasabni** [plaidml/tpp-mlir#246]
**Test case** - `add` of two 2D tensors that are initialized to 1.0 by default in `tpp-run`

```
#map = affine_map<(d0, d1) -> (d0, d1)>

!tensor_t  = tensor<4x2xf32> // N,H,W,Ic

func.func @entry(%input1 : !tensor_t,
                 %input2 : !tensor_t) -> !tensor_t {
    %1487 = tensor.empty() : !tensor_t
    %1488 = linalg.generic {
            indexing_maps = [#map, #map, #map], 
            iterator_types = ["parallel", "parallel"]
        }
        ins(%input1, %input2 : !tensor_t,  !tensor_t) 
        outs(%1487 : !tensor_t) {
            ^bb0(%in1: f32, %in2: f32, %out: f32):
                %x = arith.addf %in1, %in2 : f32
                linalg.yield %x : f32
    } -> !tensor_t

    return %1488 : !tensor_t
}
```

**Command**
```
bin/tpp-opt add_2d.mlir -empty-tensor-to-alloc-tensor  -one-shot-bufferize="bufferize-function-boundaries allow-return-allocs function-boundary-type-conversion=identity-layout-map"  -drop-equivalent-buffer-results -finalizing-bufferize -canonicalize -expand-strided-metadata -lower-affine | bin/tpp-run -n 10 -print -e entry -entry-point-result=void -shared-libs=%llvmlibdir/libmlir_c_runner_utils%shlibext,%tpplibdir/libtpp_c_runner_utils%shlibext
```

**Expected output**

```
( 2, 2 )
( 2, 2 )
( 2, 2 )
( 2, 2 )
```

**Actual output**
```
( 1, 1 )
( 1, 1 )
( 1, 1 )
( 1, 1 )
```

**Incorrect `main` code generated by `tpp-run`**
```
func.func @entry() {
    %0 = memref.get_global @__wrapper_0 : memref<4x2xf32>
    %1 = memref.get_global @__wrapper_1 : memref<4x2xf32>
    %2 = call @_entry(%0, %1) : (memref<4x2xf32>, memref<4x2xf32>) -> memref<4x2xf32>
    %cst = arith.constant -1.000000e+00 : f32
    %c4 = arith.constant 4 : index
    %c0 = arith.constant 0 : index
    %c1 = arith.constant 1 : index
    scf.for %arg0 = %c0 to %c4 step %c1 {
      %8 = vector.transfer_read %1[%arg0, %c0], %cst : memref<4x2xf32>, vector<2xf32>
      vector.print %8 : vector<2xf32>
    }
    ...
```
Notice that `%2` is not used in `transfer_read` and `print`.

**Expected `main` code in `tpp-run`**
```
func.func @entry() {
    %0 = memref.get_global @__wrapper_0 : memref<4x2xf32>
    %1 = memref.get_global @__wrapper_1 : memref<4x2xf32>
    %2 = call @_entry(%0, %1) : (memref<4x2xf32>, memref<4x2xf32>) -> memref<4x2xf32>
    %cst = arith.constant -1.000000e+00 : f32
    %c4 = arith.constant 4 : index
    %c0 = arith.constant 0 : index
    %c1 = arith.constant 1 : index
    scf.for %arg0 = %c0 to %c4 step %c1 {
      %8 = vector.transfer_read %2[%arg0, %c0], %cst : memref<4x2xf32>, vector<2xf32>
      vector.print %8 : vector<2xf32>
    }
    ....
```
Notice that `%2` is used in `transfer_read` now.

**Bug and fix**
Bug is in line [ auto funcType = main.getFunctionType();](https://github.com/plaidml/tpp-mlir/blob/2e42905bd3c6be5e0227817847e264e43444696f/tpp-run/MLIRBench.cpp#L147)
```
diff --git a/tpp-run/MLIRBench.cpp b/tpp-run/MLIRBench.cpp
index 8bfe531..d13e6c7 100644
--- a/tpp-run/MLIRBench.cpp
+++ b/tpp-run/MLIRBench.cpp
@@ -144,7 +144,7 @@ Value MLIRBench::callKernel(llvm::SmallVector<llvm::StringRef> &list) {
   // Call the Kernel, making sure to set the result to either the return value
   // or the last argument, if the return is void.
   Value result;
-  auto funcType = main.getFunctionType();
+  auto funcType = kernel.getFunctionType();
   if (funcType.getNumResults() == 0) {
     builder.create<func::CallOp>(unkLoc, kernel, kernelArgs);
     result = kernelArgs.back();
@@ -318,6 +318,8 @@ LogicalResult MLIRBench::finalize() {
   passManager.addNestedPass<func::FuncOp>(createCanonicalizerPass());
   passManager.addPass(createReconcileUnrealizedCastsPass());
```

**Comment by nhasabni** [comment]
@rengolin Please let me if this is indeed a bug - I believe `-print` option is expected to print output of a kernel. If so, I can submit a PR. Thanks.

**Comment by rengolin** [comment]
Yes! It is a bug! Thanks for catching it. Probably a by-product of moving the code from `tpp-run.cpp` to `MLIRBench.cpp`.

I wondered why this was "working" so far, and looked in the benchmarks, which only have `void` return for the kernels, so it always "matched".

**Comment by nhasabni** [comment]
Thanks for confirming. I was also wondering why does [this test](https://github.com/plaidml/tpp-mlir/blob/2e42905bd3c6be5e0227817847e264e43444696f/test/Benchmarks/mlp.mlir#L14) return output in `%output`. Explains now.

**Comment by rengolin** [comment]
Fixed by #248
